python/pyspark/sql/streaming/listener.py

```diff
@@ -75,6 +75,11 @@ def spark(self) -> Optional["SparkSession"]:  # type: ignore[name-defined] # noq
         else:
             return None
 
+    @spark.setter
+    def spark(self, session):
+        # For backward compatibility
+        self._sparkSession = session
```

Comment:
Do we need to update `_set_spark_session` to not override if the user has already set a spark session?

Reply:
SGTM